typescript/cli/src/validator/preFlightCheck.ts

```diff
@@ -0,0 +1,136 @@
+import { MerkleTreeHook__factory } from '@hyperlane-xyz/core';
+import { HyperlaneCore, S3Validator } from '@hyperlane-xyz/sdk';
+import { Address, eqAddress } from '@hyperlane-xyz/utils';
+
+import { CommandContext } from '../context/types.js';
+import { errorRed, logBlue, logGreen, warnYellow } from '../logger.js';
+
+export const checkValidatorSetup = async (
+  context: CommandContext,
+  chain: string,
+  validators: Set<Address>,
+) => {
+  const { multiProvider, registry } = context;
+
+  const addresses = await registry.getAddresses();
+
+  const core = HyperlaneCore.fromAddressesMap(addresses, multiProvider);
+
+  const validatorAnnounce = core.getContracts(chain).validatorAnnounce;
+  const merkleTreeHook = MerkleTreeHook__factory.connect(
+    addresses[chain].merkleTreeHook,
+    multiProvider.getProvider(chain),
+  );
+
+  let merkleTreeLatestCheckpointIndex: number | undefined;
+  try {
+    const [_, latestCheckpointIndex] = await merkleTreeHook[
+      'latestCheckpoint()'
+    ]();
+
+    merkleTreeLatestCheckpointIndex = latestCheckpointIndex;
+    logBlue(
+      `\nLatest check point index of incremental merkle tree: ${merkleTreeLatestCheckpointIndex}\n`,
+    );
+  } catch (err) {
+    warnYellow(
```

Comment:
Should we fail if this happens?

Reply:
As in make this an error instead of a warning?

Reply:
ah I meant putting `err` in the message